Sources/SwiftFusion/Geometry/Pose2.swift

```diff
@@ -1,34 +1,223 @@
-/// Pose2 class is the new Swift type for the SE(2) manifold of 2D Euclidean
-/// Poses
-public struct Pose2: Equatable, Differentiable, KeyPathIterable, TangentStandardBasis {
-  public var t_: Point2
-  public var rot_: Rot2
+import TensorFlow
 
+/// SE(2) Lie group of 2D Euclidean Poses.
+///
+/// We adopt the conventions in [1] and [2] for the tangent space of this manifold and for
+/// derivatives of functions to/from this manifold. In particular:
+///
+///   1. The tangent space at every point is the Lie algebra so(2).
```

Comment:
Whoops, typo, I did mean se(2).

Your argument about making `TangentVector == Vector3` is convincing. I'll change it. This will require me to actually create the `Vector*` types, and I would like to do that in a separate PR so that this one doesn't get way too big and complicated, so I'll work on creating and merging that before updating the PR to use it.

(I'm particularly convinced by the thing you said during our video chat about how writing a data structure for a linearized factor graph is easy when you make all of the tangent vectors of all the different types be compatible "flat" vectors. I didn't really see the connection between this and that until I actually started playing with a "dumb" implementation of a NonlinearFactorGraph, but now I see :))

Reply:
Awesome to be on the same page - it will also allow to easily port many unit tests from GTSAM. But the Pose2 exercise shows how awesome the synthesized derivatives approach is!!! I am very impressed by that.